File: cosmosis/plotting/plot_demo_3.py
#!/usr/bin/env python
import matplotlib
matplotlib.use("Agg")
import pylab
import numpy as np


data = np.loadtxt("demo3_output.txt").T

r_T = data[0]
log_like = data[1]

#convert to like from log-like and normalize
like = np.exp(log_like)
like /= like.max()

#make plot
pylab.plot(r_T, like)
pylab.xlabel("Tensor ratio r")
pylab.ylabel("Likelihood")
pylab.grid()

#Save
pylab.savefig("plots/demo3.png")

File: cosmosis/plotting/plot_demo_3.r
#!/usr/bin/env Rscript

library(lattice)

data <- read.table("demo3_output.txt")
names(data) <- c("r", "loglike")

#convert to like from log-like and normalize
data$like <- exp(data$loglike)
norm <- data$like/max(data$like)
data$like <- data$like/norm

# Make the output directory if needed.
if (! file.exists("plots"))
   dir.create("plots")

#make plot
png("plots/demo3.png")
xyplot(like~r, data, xlab="Tensor ratio r", ylab="Likelihood", grid=TRUE, type="l")
invisible(dev.off())
